packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GetOauthTokenTask.java

```diff
@@ -0,0 +1,87 @@
+package io.flutter.plugins.googlesignin;
+
+import android.accounts.Account;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+import com.google.android.gms.auth.GoogleAuthUtil;
+import java.util.List;
+
+/** Android inbuilt solution for calling async tasks */
+class GetOauthTokenTask extends AsyncTask<GetOauthTokenTask.Request, Void, String> {
+
+  private final OnTokenListener listener;
+  @Nullable private Exception error;
+
+  public GetOauthTokenTask(@NonNull OnTokenListener listener) {
+    this.listener = listener;
+  }
+
+  @Override
+  protected String doInBackground(GetOauthTokenTask.Request... requests) {
+    if (requests.length == 1) {
```

Comment:
If we don't really handle request counts other than 1, we should probably throw an `IllegalArgumentException` rather than fail silently.